### ISSUE TITLE
Error after update to 1.4.

### ISSUE THREAD

**Issue by a-hausmann** [minad/vertico#380]
I updated to Vertico 1.4 today, and now receiving error message, "Error in post-command-hook (vertico--exhibit): (invalid-function with-memoization)" upon M-x. The completion is completely broken. Have tried deleting the vertico directory and reinstalling, that didn't help.
Not sure what is wrong as the post-command-hook is used only in vertico--setup, and I don't see anything wrong with that code, though the cl-defgeneric baffles me.

**Comment by a-hausmann** [comment]
My config for completion...
[completion-config.txt](https://github.com/minad/vertico/files/11963727/completion-config.txt)

**Comment by minad** [comment]
You seem to be missing the Compat package which provides with-memoization. I suggest you reinstall Compat and Vertico. Which Emacs versiom do you use?

**Comment by a-hausmann** [comment]
I'm on version 27.1, and compat-29.1.4.1 was installed yesterday when I upgraded to vertico-1.4. OTOH, I note that I still had both compat-28.1.2.0 and marginalia-20220721.1833 still in elpa directory though I had updated both to latest version.  Deleting all and reinstalling has solved the issue. Thanks for your assistance...I'm still not good enough to figure out these obtuse errors when I get them.
